### PR TITLE
fix: [BT-1616]: speed up graphql schema load time

### PR DESCRIPTION
**Pull request by rickmarry**
reduce the number of times we load the schema files from 3 to 2

## Describe your changes
Very simple refactor that results in calling loadSchemaForEnv 2 times instead of 3.. 
The outcome is identical and we reduce load time by about 29%  

graphql_new_faster_jps_-_JProfiler_13_0_2_and_graphql_orig_slow_jps_-_JProfiler_13_0_2

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36218)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36218)

File: 400-rest/src/main/java/software/wings/graphql/provider/GraphQLProvider.java [hunk 1]
```diff
@@ -56,24 +56,20 @@ public class GraphQLProvider implements QueryLanguageProvider<GraphQL> {
 
   @Inject
   public void init() {
-    if (privateGraphQL == null) {
-      String[] allPaths = new String[] {GRAPHQL_SCHEMA_PRIVATE_DIRECTORY_PATH, GRAPHQL_SCHEMA_PUBLIC_DIRECTORY_PATH};
-      privateGraphQL = getGraphQL(allPaths);
+    TypeDefinitionRegistry typeDefinitionRegistry = new TypeDefinitionRegistry();
+    if (publicGraphQL == null) {
+      publicGraphQL = getGraphQL(GRAPHQL_SCHEMA_PUBLIC_DIRECTORY_PATH, typeDefinitionRegistry);
```
Comment (rijajoo) [inline 400-rest/src/main/java/software/wings/graphql/provider/GraphQLProvider.java:61]: Shouldn't the privateGraphQL contain GRAPHQL_SCHEMA_PUBLIC_DIRECTORY_PATH as well?

**Review by rijajoo — CHANGES_REQUESTED** [review]

File: 400-rest/src/main/java/software/wings/graphql/provider/GraphQLProvider.java (hunk 1 above)
Comment (rickmarry) [reply 400-rest/src/main/java/software/wings/graphql/provider/GraphQLProvider.java:61]: It does.. the typeDefinitionRegistry should accumulate it when we get the public one

**Review by rijajoo — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36218#-:-N9Bao4dETbnKoQidtF8:bnfp4nl)* status: 1. unresolved discussions.

**Review by ankit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36218)* status: 1. unresolved discussions.

**Review by rickmarry — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36218)* status: 1. unresolved discussions.

**Review by rickmarry — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36218)* status: 1. unresolved discussions. 

___
*[`400-rest/src/main/java/software/wings/graphql/provider/GraphQLProvider.java` line 61 at r1](https://reviewable.io/reviews/harness/harness-core/36218#-N92vIUPZ5OPXe2oRlLH-r1-61:-N9C_TdmBYYRcC4AZ40v:b32wtd) ([raw file](https://github.com/harness/harness-core/blob/85d748ea06bd041013d54adeeaf2da55508a9092/400-rest/src/main/java/software/wings/graphql/provider/GraphQLProvider.java#L61)):*
<details><summary><i>Previously, rickmarry (Rick Marry) wrote…</i></summary><blockquote>

It does contain it. The typeDefinitionRegistry will accumulate it when we get the public one.  Prior to this change, typeDefinitionRegistry was a local var.. now we pass it in so that when it gets to do the private one, it also gets what public has accumulated
</blockquote></details>

ok

**Review by rijajoo — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36218)* status: :shipit: complete! ✓

**Pull request merged**